cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -465,4 +467,31 @@ int32 locate_utf8_utf8_int32(int64 context, const char* sub_str, int32 sub_str_l
   return 0;
 }
 
+#define CAST_VARCHAR(IN_TYPE, ARROW_TYPE)                                            \
+  FORCE_INLINE                                                                       \
+  const char* castVARCHAR_##IN_TYPE##_int64(int64 context, IN_TYPE value, int64 len, \
+                                            int32 * out_len) {                       \
+    arrow::internal::StringFormatter<arrow::ARROW_TYPE> formatter;                   \
```

Comment:
is len = 0 an error (or) expected?

Reply:
len = 0 would return empty string. Added a check for Len < 0